huaweicloud/services/mrs/resource_huaweicloud_mapreduce_job.go

```diff
@@ -0,0 +1,525 @@
+package mrs
+
+import (
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/huaweicloud/golangsdk"
+	"github.com/huaweicloud/golangsdk/openstack/mrs/v2/jobs"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+func ResourceMRSJobV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMRSJobV2Create,
+		Read:   resourceMRSJobV2Read,
+		Delete: resourceMRSJobV2Delete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceMRSClusterSubResourceImportState,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[A-Za-z0-9_-]{1,64}$"),
+					"The name consists of 1 to 64 characters, which only letters, digits, hyphens (-) and "+
+						"underscores (_) are allowed."),
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Flink", "HiveSql", "HiveScript", "MapReduce", "SparkSubmit", "SparkSql", "SparkScript",
```

Comment:
defined as a constant, it is more convenient to use, such as in test : resource.TestCheckResourceAttr(resourceName, "type", "SparkSubmit"),